File: python/Deck.py
 # Deck.py

import random
from Card import Card
from typing import List

class Deck:
    
    # deck constructor
    def __init__(self):
        self.cards = []
        suits = "DCHS"
        for s in suits:
            for rank in range(2, 15): # 2-14
                self.cards.append(Card(rank, s))

    # method to shuffle deck
    def shuffle(self):
        random.shuffle(self.cards)

    # method to deal hand of six
    def deal_six(self) -> List[Card]:
        hand = self.cards[:6]
        del self.cards[:6]
        return hand

    # method for printing deck - when false
    def print_deck(self, single_line: bool):
        for i in range(len(self.cards)):
            card = self.cards[i]
            print(card, end="")
            end_of_row = (not single_line) and ((i + 1) % 13 == 0)
            if end_of_row:
                print()
            elif i + 1 < len(self.cards):
                print(" ", end="")
        print()    

File: java/Deck.java
import java.util.*;

public class Deck {

	private List <Card> cards; //all cards in the deck
	
	// Deck constructor - inorder
	public Deck() {
		cards = new ArrayList<>();
		char[] suits = {'D','C','H','S'};
		
		for (int i = 0; i < suits.length; i++) {
			char suit = suits[i];
			for (int rank = 2; rank <= 14; rank++) {
				cards.add(new Card(rank,suit));
			}
		}
	}
	
	//method to shuffle deck
	public void shuffle() {
		Collections.shuffle(cards, new Random());
	}

	// method to deal hand
	public List<Card> dealSix() {
		List<Card> hand = new ArrayList<>(6);
		for (int i = 0; i < 6; i++) {
			hand.add(cards.remove(0));
		}
		return hand;
	}

	// Method for printing deck - when false it will print deck
	public void printDeck(boolean singleLine) {
		for (int i = 0; i < cards.size(); i++) {
			System.out.print(cards.get(i));
			boolean endOfRow = !singleLine && ((i+1) % 13 == 0);
		
			if (endOfRow) {
				System.out.println();
			} else if (i + 1 < cards.size()) {
				System.out.print(" ");
			}
		}
		System.out.println();
	}
}			



